2022/en/red-hat-developers-blogs/josdk-part-3.md

```diff
@@ -0,0 +1,360 @@
+        .withNewSpec()
+            .addNewPort()
+                .withName("http")
+                .withPort(8080)
+                .withNewTargetPort().withIntVal(8080).endTargetPort()
+            .endPort()
+            .withSelector(labels)
+            .withType("ClusterIP")
+        .endSpec()
+.build());
+```
+
+The `Ingress` is slightly more complex because it actually depends on
+which [`Ingress` controller](https://kubernetes.io/fr/docs/concepts/services-networking/ingress/) is deployed on your
+cluster. In this example, you'll be configuring the `Ingress` specifically for
+the [NGINX controller](https://kubernetes.github.io/ingress-nginx/). If your cluster uses a different controller, the
+configuration will be different:
+
+```java,noformat
+final var metadata = createMetadata(exposedApp, labels);
+metadata.setAnnotations(Map.of(
+    "nginx.ingress.kubernetes.io/rewrite-target", "/",
+    "kubernetes.io/ingress.class", "nginx"
+));
+
+client.network().v1().ingresses().createOrReplace(new IngressBuilder()
+    .withMetadata(metadata)
+    .withNewSpec()
+        .addNewRule()
+            .withNewHttp()
+                .addNewPath()
+                    .withPath("/")
+                    .withPathType("Prefix")
+                    .withNewBackend()
+                        .withNewService()
+                            .withName(metadata.getName())
+                            .withNewPort().withNumber(8080).endPort()
+                        .endService()
+                    .endBackend()
+                .endPath()
+            .endHttp()
+        .endRule()
+    .endSpec()
+.build());
+```
+
+And that's about it for this very simple reconciliation algorithm! The only you need to do is return an
+`UpdateControl` to JOSDK know what to do with your CR once it's reconciled. In this case, you're not modifying the CR in
+any form, so you simply return `UpdateControl.noUpdate()`. JOSDK now knows that it doesn't need to send an updated
+version of your CR to the cluster and can update its internal state accordingly.
+
+At this point, your operator should still be running using the Quarkus dev mode. If you create an `ExposedApp`
+resource and apply it to the cluster using `kubectl apply`, your operator should now create the associated resources as
+you can see running the following command:
+
+```shell
+kubectl get all -l app.kubernetes.io/name=<name of your CR>
+```
+
+If everything worked well, you should indeed see that a `Pod`, a `Service`, a `Deployment` and a `ReplicaSet` have all
+been created for your application. `Ingresses` are not part of the resources that are displayed by a `kubectl get all`
+so you'd need a separate command to check on your `Ingress`:
+
+```shell
+kubectl get ingresses.networking.k8s.io -l app.kubernetes.io/name=<name of your CR>
+```
+
+In our case, we created an `hello-quarkus` `ExposedApp` that exposes a simple "Hello World" Quarkus application and got
```

Comment:
```suggestion
In our case, we created a `hello-quarkus` `ExposedApp` that exposes a simple "Hello World" Quarkus application and got
```
not sure of that but just in case